mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/model/impl/MessageParser.java

```diff
@@ -194,13 +195,24 @@ private boolean isInline(Optional<ContentDispositionField> contentDispositionFie
     }
 
     private boolean isAttachment(Entity part, Context context) {
-        if (context == Context.BODY && isTextPart(part)) {
+        boolean bodyTextCriterion = context == Context.BODY && isTextPart(part);
+
+        boolean attachmentDispositionCriterion = getContentDispositionField(part)
+            .map(ContentDispositionField::getDispositionType)
+            .map(dispositionType -> dispositionType.toLowerCase(Locale.US))
```

Comment:
why do we have to do string handling here ? Can't we rely on ContentDispositionField for that ?

Reply:
That was done before.

Moreover I don't see why we should correlate external library changes to a fix.

Reply:
1. mime4j is part of james
2. I don't know ContentDispositionField, so I was only asking if it's doable
3. If we don't care about mime4j being a great lib, who will ? I'm sad when we fallback to javax.mail or workaround limitation